test/FullRange.t.sol

```diff
@@ -0,0 +1,873 @@
+
+        assertEq(TestERC20(token0).balanceOf(address(this)), currBalance0 - 150);
+        assertEq(TestERC20(token1).balanceOf(address(this)), currBalance1 - 150);
+
+        assertEq(UniswapV4ERC20(liquidityToken).balanceOf(address(this)), 150);
+    }
+
+    function testAddLiquiditySucceedsWithFee() public {
+        manager.initialize(feeKey, SQRT_RATIO_1_1);
+
+        uint256 currBalance0 = TestERC20(token0).balanceOf(address(this));
+        uint256 currBalance1 = TestERC20(token1).balanceOf(address(this));
+
+        fullRange.addLiquidity(address(token0), address(token1), 3000, 100, 100, address(this), 12329839823);
+
+        (,,,,,, address liquidityToken) = fullRange.poolToInfo(feeId);
+
+        assertEq(UniswapV4ERC20(liquidityToken).balanceOf(address(this)), 100);
+
+        fullRange.addLiquidity(address(token0), address(token1), 3000, 50, 50, address(this), 12329839823);
+
+        assertEq(TestERC20(token0).balanceOf(address(this)), currBalance0 - 150);
+        assertEq(TestERC20(token1).balanceOf(address(this)), currBalance1 - 150);
+
+        assertEq(UniswapV4ERC20(liquidityToken).balanceOf(address(this)), 150);
+
+        // check pool position state
+        (
+            uint128 liquidity,
+            uint256 feeGrowthInside0LastX128,
+            uint256 feeGrowthInside1LastX128,
+            uint128 tokensOwed0,
+            uint128 tokensOwed1,
+            ,
+        ) = fullRange.poolToInfo(feeId);
+
+        assertEq(liquidity, 150);
+        // TODO: make sure 0 is correct
+        assertEq(feeGrowthInside0LastX128, 0);
+        assertEq(feeGrowthInside1LastX128, 0);
+        assertEq(tokensOwed0, 0);
+        assertEq(tokensOwed1, 0);
+    }
+
+    function testAddLiquidityWithDiffRatiosAndNoFee() public {
+        manager.initialize(key, SQRT_RATIO_1_1);
+
+        uint256 currBalance0 = TestERC20(token0).balanceOf(address(this));
+        uint256 currBalance1 = TestERC20(token1).balanceOf(address(this));
+
+        fullRange.addLiquidity(address(token0), address(token1), 0, 100, 100, address(this), 12329839823);
+
+        (,,,,,, address liquidityToken) = fullRange.poolToInfo(id);
+
+        assertEq(UniswapV4ERC20(liquidityToken).balanceOf(address(this)), 100);
+
+        fullRange.addLiquidity(address(token0), address(token1), 0, 50, 25, address(this), 12329839823);
+
+        // even though we desire to deposit more token0, we cannot, since the ratio is 1:1
+        assertEq(TestERC20(token0).balanceOf(address(this)), currBalance0 - 125);
+        assertEq(TestERC20(token1).balanceOf(address(this)), currBalance1 - 125);
+
+        assertEq(UniswapV4ERC20(liquidityToken).balanceOf(address(this)), 125);
+    }
+
+    function testAddLiquidityWithDiffRatiosAndFee() public {
+        manager.initialize(feeKey, SQRT_RATIO_1_1);
+
+        uint256 currBalance0 = TestERC20(token0).balanceOf(address(this));
+        uint256 currBalance1 = TestERC20(token1).balanceOf(address(this));
```

Comment:
probably more aptly named `previousBalance` since by the time we're referencing it, it's not current.